Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs

```diff
@@ -199,16 +202,41 @@ private void SubtractContents(MicrowaveComponent component, FoodRecipePrototype
                 {
                     foreach (var item in component.Storage.ContainedEntities)
                     {
-                        var metaData = MetaData(item);
-                        if (metaData.EntityPrototype == null)
+                        string? itemID = null;
+
+                        // If an entity has a stack component, use the stacktype instead of prototype id
+                        if (TryComp<StackComponent>(item, out var stackComp))
+                        {
+                            itemID = _prototype.Index<StackPrototype>(stackComp.StackTypeId).Spawn;
+                        }
+                        else
+                        {
+                            var metaData = MetaData(item);
+                            if (metaData.EntityPrototype == null)
+                            {
+                                continue;
+                            }
+                            itemID = metaData.EntityPrototype.ID;
+                        }
+
+                        if (itemID != recipeSolid.Key)
                         {
                             continue;
                         }
 
-                        if (metaData.EntityPrototype.ID == recipeSolid.Key)
+                        if (stackComp is not null)
+                        {
+                            if (stackComp.Count == 1)
+                            {
+                                _container.Remove(item, component.Storage);
+                            }
+                            _stack.Use(item, 1, stackComp);
+                            break;
+                        }
+                        else
                         {
                             _container.Remove(item, component.Storage);
-                            EntityManager.DeleteEntity(item);
+                            Del(item);
                             break;
                         }
```

Comment:
now, the last thing that irks me about this,  couldnt you just merge this into a single if else and just repeat the itemID != recipeSolid.Key check?

Reply:
It would work but I think the current structure makes more sense:
First we check the entity and get an ID
Then check if that id is the one we are currently looking for, if not, iterate to the next entity
Then remove the entity or decrement the stack component count

An outer loop is counting down the amount of solids needed to be removed, so every iteration of this loop will only remove 1 entity or decrement the stack by 1.

Reply:
eh, its just a shorter version of basically the same thing but i wont block this over that